extensions/data-plane-transfer/data-plane-transfer-core/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/core/DataPlaneTransferCoreExtension.java

```diff
@@ -14,22 +14,81 @@
 import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
 import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.eclipse.dataspaceconnector.spi.transfer.edr.EndpointDataReferenceTransformer;
+import org.eclipse.dataspaceconnector.transfer.dataplane.core.proxy.DataPlaneProxyManagerImpl;
+import org.eclipse.dataspaceconnector.transfer.dataplane.core.proxy.DataPlaneProxyTokenGeneratorImpl;
 import org.eclipse.dataspaceconnector.transfer.dataplane.core.security.NoopDataEncrypter;
 import org.eclipse.dataspaceconnector.transfer.dataplane.spi.security.DataEncrypter;
+import org.eclipse.dataspaceconnector.transfer.dataplane.spi.security.DataPlaneProxyManager;
 
-@Provides({DataEncrypter.class})
+import java.security.PrivateKey;
+import java.security.interfaces.ECPrivateKey;
+import java.util.concurrent.TimeUnit;
+
+@Provides({DataEncrypter.class, DataPlaneProxyManager.class})
 public class DataPlaneTransferCoreExtension implements ServiceExtension {
 
+    @EdcSetting
+    private static final String DATA_PLANE_PUBLIC_API_ENDPOINT = "edc.transfer.dataplane.sync.endpoint";
+
+    @EdcSetting
+    private static final String DATA_PLANE_TOKEN_VALIDITY_SECONDS = "edc.transfer.dataplane.sync.token.validity";
+    private static final long DEFAULT_DATA_PLANE_TOKEN_VALIDITY_SECONDS = TimeUnit.MINUTES.toSeconds(10);
+
+    @EdcSetting
+    private static final String TOKEN_SIGNER_PRIVATE_KEY_ALIAS = "edc.transfer.dataplane.token.signer.privatekey.alias";
+
     @Override
     public String name() {
         return "Data Plane Transfer Core";
     }
 
     @Override
     public void initialize(ServiceExtensionContext context) {
-        context.registerService(DataEncrypter.class, new NoopDataEncrypter());
+        var tokenValidity = context.getSetting(DATA_PLANE_TOKEN_VALIDITY_SECONDS, DEFAULT_DATA_PLANE_TOKEN_VALIDITY_SECONDS);
+        var endpoint = context.getConfig().getString(DATA_PLANE_PUBLIC_API_ENDPOINT);
+
+        var privateKeyAlias = context.getConfig().getString(TOKEN_SIGNER_PRIVATE_KEY_ALIAS);
+        var tokenSigner = createTokenSigner(context.getService(PrivateKeyResolver.class), privateKeyAlias);
+        var tokenGenerationService = new TokenGenerationServiceImpl(tokenSigner);
+
+        var encryper = new NoopDataEncrypter();
+        context.registerService(DataEncrypter.class, encryper);
+
+        var dataPlaneProxyTokenGenerator = new DataPlaneProxyTokenGeneratorImpl(context.getTypeManager(), encryper, tokenGenerationService, tokenValidity);
+        var proxyManager = new DataPlaneProxyManagerImpl(endpoint, dataPlaneProxyTokenGenerator);
+        context.registerService(DataPlaneProxyManager.class, proxyManager);
+        context.registerService(EndpointDataReferenceTransformer.class, proxyManager);
+    }
+
+    /**
+     * Create token signer from private key.
+     */
+    private JWSSigner createTokenSigner(PrivateKeyResolver resolver, String pkAlias) {
```

Comment:
nice catch, it took me some time but I managed to merge the implementation and took the opportunity to put the JwtDecorator/JwtDecoratorRegistry stuff in token generation lib instead of oauth2.